VCEntities/VCEntities/backup/container/jwe/JwePasswordProtectionMethod.swift

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import Foundation
+import VCCrypto
+import VCToken
+
+enum JwePasswordProtectionError : Error {
+    case invalidContentType
+}
+
+public struct JwePasswordProtectionMethod : BackupProtectionMethod {
+    
+    private struct Constants {
+        static let Algorithm = "PBES2-HS512+A256KW"
+        static let Method = "A256CBC-HS512"
+        static let SaltLength = 8
+        static let SaltIterations = UInt(100 * 1000)
+    }
+    
+    let password: String
+    
+    public init(password:String) {
+        self.password = password
```

Comment:
just curious: do we need to remove the password from memory like we do with keys?

Reply:
I'm not sure. Unlike the private keys used for signing, etc., this _is_ a value which we show to the user (and receive from them) so even changing this to remove the password after we're done with it (which would require some refactoring, since I'm guessing `String`s are immutable in Swift), then there will be other copies of it about the place: in a screen buffer when we show them the user the password, and in a text buffer after they've typed it in